src/processor/operator/hash_join/join_hash_table.cpp

```diff
@@ -42,9 +42,12 @@ void JoinHashTable::append(const std::vector<ValueVector*>& vectorsToAppend) {
     // TODO(Guodong): use compiling information to remove the for loop.
     auto numTuplesToAppend = 1;
     for (auto i = 0u; i < numKeyColumns; i++) {
-        numTuplesToAppend *= (vectorsToAppend[i]->state->isFlat() ?
-                                  1 :
-                                  vectorsToAppend[i]->state->selVector->selectedSize);
+        // At most one unFlat key data chunk. If there are multiple unFlat key vectors, they must
+        // share the same state.
+        if (!vectorsToAppend[i]->state->isFlat()) {
```

Comment:
This logic is correct but still garbage in the sense that this should be a shared logic implemented in factorized table. Go back and update comment and issue.